File: day02/day2a.f90
PROGRAM Day2
  IMPLICIT NONE
  INTEGER(4) :: iErr,step,depth,horz
  CHARACTER(LEN=7) :: dir

  depth = 0
  horz = 0
  OPEN(unit=98,file="input.dat")
  DO WHILE(.TRUE.)
    READ(98,*,iostat=iErr) dir,step
    IF (iErr /= 0) EXIT
    SELECT CASE(dir)
    CASE("up")
      depth = depth - step
    CASE("down")
      depth = depth + step
    CASE("forward")
      horz = horz + step
    CASE DEFAULT
      STOP 123
    ENDSELECT
  ENDDO
  WRITE(*,*) depth*horz
ENDPROGRAM Day2

File: 2021/fortran/day02/day2a.py

depth = 0
horz = 0
with open('input.txt','r') as inp:
    for line in inp:
        instr,step = line.split()
        step = int(step)
        if instr=="down":
            depth += step
        elif instr=="up":
            depth -= step
        elif instr=="forward":
            horz += step

print(depth*horz)

